### PR TITLE
Initial commit of fabric version

### PR DESCRIPTION
**Pull request by lnobach**
- P4 sources
- Wireshark dissectors for cpu headers

### CODE REVIEW

File: fabric/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+
+P4 CORD Service Edge
+====================
+
+This repository contains a P4 program that implements
+a residential network access service data plane. The residential access data
+plane is desgined for the requirements of Deutsche Telekom and includes PPPoE
```
Comment (ccascone) [inline fabric/README.md]: typo

File: README.md [hunk 2]
```diff
@@ -6,3 +6,8 @@ BNG/PPPoE P4 software
 
 The directory `generic` contains a cross-platform P4 implementation 
 for devices intended to be attached to a leaf switch, tested against bmv2.
+
+## Fabric
+
+The directory `fabric` contains a P4 implementation integrated into a CORD
```
Comment (ccascone) [inline README.md]: I believe in the CORD community the wording "CORD leaf switch" is understood as a switch that is part of a leaf-spine network controlled by Trellis. Can this program be controlled by Trellis? Has it been tested in conjunction with other Trellis-controller switches (e.g. spine)? If not I would reword this to something like "integrated into a leaf switch of a CORD-like leaf-spine topology"

**Review by ccascone — CHANGES_REQUESTED** [review]

File: fabric/README.md (hunk 1 above)
Comment (ccascone) [reply fabric/README.md]: desgined -> designed

**Comment by lnobach** [comment]
The issues are fixed now.
